docs/sdk/src/reference_docs/frame_offchain_workers.rs

```diff
@@ -0,0 +1,115 @@
+//! # Offchain Workers
+//!
+//! This reference document explains how offchain workers work in Substrate and FRAME. The main
+//! focus is upon FRAME's implementation of this functionality. Nonetheless, offchain workers are a
+//! Substrate-provided feature and can be used with possible alternatives to [`frame`] as well.
+//!
+//! Offchain workers are a commonly misunderstood topic, therefore we explain them bottom-up,
+//! starting at the fundamentals and then describing the developer interface.
+//!
+//! ## Context
+//!
+//! Recall from [`crate::reference_docs::wasm_meta_protocol`] that the node and the runtime
+//! communicate with one another via host functions and runtime APIs. Many of these interactions
+//! contribute to the actual state transition of the blockchain. For example [`sp_api::Core`] is the
+//! main runtime API that is called to execute new blocks.
+//!
+//! Offchain workers are in principle not different in any way: It is a runtime API exposed by the
+//! wasm blob ([`sp_offchain::OffchainWorkerApi`]), and the node software calls into it when it
+//! deems fit. But, crucially, this API call is different in that:
+//!
+//! 1. It can have no impact on the state ie. it is _OFF (the) CHAIN_. If any state is altered
+//!    during the execution of this API call, it is discarded.
+//! 2. It has access to an extended set of host functions that allow the wasm blob to do more. For
+//!    example, call into HTTP requests.
+//!
+//! > The main way through which an offchain worker can interact with the state is by submitting an
+//! > extrinsic to the chain. This is the ONLY way to alter the state from an offchain worker.
+//! > [`pallet_example_offchain_worker`] provides an example of this.
+//!
+//!
+//! Given the "Off Chain" nature of this API, it is important to remember that calling this API is
+//! entirely optional. Some nodes might call into it, some might not, and it would have no impact on
+//! the execution of your blockchain because no state is altered no matter the execution of the
+//! offchain worker API.
+//!
+//! Substrate's CLI allows some degree of configuration about this, allowing node operators to
+//! specify when they want to run the offchain worker API. See
+//! [`sc_cli::RunCmd::offchain_worker_params`].
+//!
+//! ## Nondeterministic Execution
+//!
+//! Needless to say, given the above description, the code in your offchain worker API can be
+//! nondeterministic, as it is executed at unknown times, by unknown nodes, and has no impact on the
```

Comment:
```suggestion
//! nondeterministic, as it is not part of the blockchain's STF, so it can be executed at unknown times, by unknown nodes, and has no impact on the
```